src/db/scheema.sql

```diff
@@ -0,0 +1,38 @@
+-- comapany database holding the employee, role and department tables
+
+DROP DATABASE IF EXISTS company_db;
+CREATE DATABASE company_db;
+
+USE company_db;
+
+CREATE TABLE `department` (
+    `id` INT NOT NULL AUTO_INCREMENT,
+    `name` VARCHAR(30) NOT NULL,
+    `createdAt` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    `updatedAt` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+    PRIMARY KEY (`id`)
+    ON DELETE CASCADE
```

Comment:
This line is causing issues. Your `ON DELETE` lines should be added to the foreign keys on the other tables, rather than here.